src/strategies/gadget.rs

```diff
@@ -365,44 +375,45 @@ mod tests {
             });
 
             composer.add_dummy_constraints();
-            //let pub_inp = composer.public_inputs().clone();
-            (composer, vec![BlsScalar::zero()])
+            vec![BlsScalar::zero()]
         }
 
         // Setup OG params.
         let public_parameters = PublicParameters::setup(CAPACITY, &mut rand::thread_rng()).unwrap();
         let (ck, vk) = public_parameters.trim(CAPACITY).unwrap();
-        let domain = EvaluationDomain::new(CAPACITY).unwrap();
 
         let (i, o) = hades();
-        let (mut composer, pi) = new_composer(i, o);
-        let mut transcript = gen_transcript();
-        // Preprocess circuit
-        let circuit = composer.preprocess(&ck, &mut transcript, &domain);
-
-        // Prove
-        let proof = composer.prove(&ck, &circuit, &mut transcript.clone());
-
-        // Verify
-        assert!(proof.verify(&circuit, &mut transcript.clone(), &vk, pi.as_slice()));
-
+        // Proving
+        let mut prover = Prover::new(b"hades_gadget_tester");
+        let pi = hades_gadget_tester(i, o, prover.mut_cs());
+        prover.preprocess(&ck).unwrap();
+        let proof = prover.prove(&ck).unwrap();
```

Comment:
Make the tests returns `Error` (as is now done in the blind bid if you need a reference).

Reply:
Is this a common practice?

Reply:
Ok, so there's a couple of things to mention related to that.

1. PLONK is using `failure` crate for error handling. The problem is that is deprecated, so I don't want to include failure here just for that. 
2. I can add a follow-up issue in respect of the error stuff that you mention and go for it once PLONK is refactored to use `anyhow` as error handling crate.

Reply:
I know `failure` is deprecated, I was the one to point it out :)

FFR, As discussed, I thought Hades was also handle the error from PLONK but apparently is something only we had in tests, so as suggested for the time being we can just use `expect` instead.

We'll refactor the tests in the future (low priority tech debt) to handle the error properly, once we have a new unified approach for error handling – possibly something not experimental such as the failure crate.